frontend/test/lit/test_quantum_control.py

```diff
@@ -130,9 +130,7 @@ def test_native_controlled_unitary():
     @qml.qnode(dev)
     # CHECK-LABEL: public @jit_native_controlled_unitary
     def native_controlled_unitary():
-        # CHECK: [[out:%.+]], [[out_ctrl:%.+]]:3 = quantum.unitary
-        # CHECK-SAME: ctrls
-        # CHECK-SAME: ctrlvals(%true, %true, %true)
+        # CHECK: [[out:%.+]] = quantum.unitary
         qml.ctrl(
```

Comment:
Why this change?

Reply:
This is now lowered to QubitUnitary with an updated matrix if it makes sense.

Reply:
Can you point out what effected that change?

Reply:
`qml.ControlledQubitUnitary` is traced and lowered as before,

```
    %4 = stablehlo.convert %arg0 : (tensor<2x2xf64>) -> tensor<2x2xcomplex<f64>>
    %out_qubits, %out_ctrl_qubits:2 = quantum.unitary(%4 : tensor<2x2xcomplex<f64>>) %1 ctrls(%2, %3) ctrlvals(%true, %true) : !quantum.bit ctrls !quantum.bit, !quantum.bit
    %5 = quantum.extract %0[ 3] : !quantum.reg -> !quantum.bit
    %6 = quantum.extract %0[ 4] : !quantum.reg -> !quantum.bit
    %7 = quantum.compbasis %out_ctrl_qubits#0, %out_ctrl_qubits#1, %out_qubits, %5, %6 : !quantum.obs
    %8 = quantum.state %7 : tensor<32xcomplex<f64>>
    %9 = quantum.insert %0[ 2], %out_qubits : !quantum.reg, !quantum.bit
    %10 = quantum.insert %9[ 0], %out_ctrl_qubits#0 : !quantum.reg, !quantum.bit
    %11 = quantum.insert %10[ 1], %out_ctrl_qubits#1 : !quantum.reg, !quantum.bit
```

However, `qml.ctrl(qml.QubitUnitary)` is no longer decomposed to `ControlledQubitUnitrary` and instead it lowers directly to `QubitUnitary` with an updated matix,

```
    %out_qubits:3 = quantum.unitary(%arg0 : tensor<8x8xcomplex<f64>>) %1, %2, %3 : !quantum.bit, !quantum.bit, !quantum.bit
```

There is no concern with the validity of results.

Reply:
Right you described the change in observable behaviour, but I'd like to understand why that change happened. What part of the PR exactly modified the decomposition behaviour, and why? 

I'm not concerned about the validity here, more so that we fully understand the code we write 😄

Reply:
Yeah, I know and I'm still investigating :smile:

Reply:
The difference here comes from the fact that, previously, `qml.ctrl(qml.QubitUnitary)` was an instance of `QCtrl` and [as this is not natively supported in lightning-qubit](https://github.com/PennyLaneAI/pennylane-lightning/blob/9b933a23fbf6b2177b393dc265ea3dcb6bde541d/pennylane_lightning/lightning_qubit/lightning_qubit.toml#L30), [the PL Controlled decomposition method was converting](https://github.com/PennyLaneAI/pennylane/blob/7f97ea735b3c43b43a0ffe34040bffaa49603268/pennylane/ops/op_math/controlled.py#L252) the gate to  `qml.ControlledQubitUnitary`. Now, we bypass the custom decomposition method of PL's Controlled and fully use Catalyst decomposition rules based on the device capabilities instead. Updating `lightning_qubit.toml` in the Lightning repo will fix this inconsistency:
```
QubitUnitary           = { properties = [ "invertible", "controllable", ] }
```